### ISSUE TITLE
As a user I want to be able to set prerequisite tasks to work on another task so that I can tell if something needs to be completed first before another thing can be worked on

### ISSUE THREAD

**Issue by aneuhold** [PointSpire/PointSpire#50]
- [ ] AC 1: Tasks can have prerequisites added to them
- [ ] AC 2: Tasks that have a prerequisite are grayed out and tasks that don't have one are white like normal (this will be able to be changed in the settings later on)
- [ ] AC 3: When a prerequisite task is completed it is checked off in any tasks that have it as a prerequisite
- [ ] AC 4: Prerequisite tasks is added to the tasks and projects
- [ ] AC 5: Make it so when a user clicks on a pre-requisite task it opens up some kind of card with details about that task. Then when they click away it goes back to where they were like normal. Kind of like a dialog box.
